web-bundle/src/main/java/com/graphhopper/customspeeds/CustomSpeedsVehicle.java

```diff
@@ -35,34 +36,35 @@ public double getMaxValidSpeed() {
 
     public final VehicleType baseVehicleType;
     public final String customVehicleName;
-    public final ImmutableMap<Long, Double> osmWayIdToCustomSpeed;
+    public final ImmutableMap<Pair<Long, Boolean>, Double> osmWayIdAndBwdToCustomSpeed;
+    public final boolean directionalCustomSpeedsProvided;
 
-    private CustomSpeedsVehicle(String customVehicleName, VehicleType baseVehicleType, ImmutableMap<Long, Double> osmWayIdToCustomSpeed) {
+    private CustomSpeedsVehicle(String customVehicleName, VehicleType baseVehicleType, ImmutableMap<Pair<Long, Boolean>, Double> osmWayIdAndBwdToCustomSpeed, boolean directionalCustomSpeedsProvided) {
         this.customVehicleName = customVehicleName;
         this.baseVehicleType = baseVehicleType;
-        this.osmWayIdToCustomSpeed = osmWayIdToCustomSpeed;
+        this.osmWayIdAndBwdToCustomSpeed = osmWayIdAndBwdToCustomSpeed;
+        this.directionalCustomSpeedsProvided = directionalCustomSpeedsProvided;
     }
 
-    public static CustomSpeedsVehicle create(String customVehicleName, ImmutableMap<Long, Double> osmWayIdToCustomSpeed) {
-        if (!validateCustomSpeeds(customVehicleName, osmWayIdToCustomSpeed)) {
+    public static CustomSpeedsVehicle create(String customVehicleName, Pair<ImmutableMap<Pair<Long, Boolean>, Double>, Boolean> osmWayIdAndBwdToCustomSpeed) {
```

Comment:
small note, I realize that the object we're passing around here has gotten kind of gross with the addition of the new direction-related info I've added. `osmWayIdAndBwdToCustomSpeed` is also not quite accurately-named here, because the structure also includes a bool for each mapping of parsed speeds representing whether or not directional speeds were provided (ie the `bwd` column was present).

One option would be to make `osmWayIdAndBwdToCustomSpeed` have an even grosser name like `osmWayIdAndBwdToCustomSpeedAndDirCustomSpeedsProvided`, which seems...not great. 

Another would be to instead create a new class to store this complex structure, but that felt kind of funny given we're basically creating the structure and then directly using it to instantiate the `CustomSpeedsVehicle` class

If you have any thoughts on how we could make this any clearer/more readable, let me know!

Reply:
Great suggestions! I like this setup to keep all of the data floating around way more organized. I've refactored everything to use the new classes/enum you suggested, with a check added to ensure that the `DirectedSpeeds` class can not be instantiated with both a non-directed speed (with direction `ALL_ACCESSIBLE`) and an explicitly directed speed (with direction `FWD` or `BWD`).

The validation that ensures directed speeds are only set in accessible directions has also been updated to ignore any speeds set in the `ALL_ACCESSIBLE` direction